test/tensor/tensor/test_tensor_assignment_operator.cpp

```diff
@@ -0,0 +1,422 @@
+//
+
+#include <boost/test/unit_test.hpp>
+#include "../fixture_utility.hpp"
+#include <boost/numeric/ublas/matrix.hpp>
+#include <boost/numeric/ublas/vector.hpp>
+
+BOOST_AUTO_TEST_SUITE(test_tensor_assignment_operator, 
+    *boost::unit_test::description("Validate tensor operator=(...)")
+    *boost::unit_test::depends_on("test_tensor_constructor")
+)
+
+
+BOOST_TEST_DECORATOR(
+    *boost::unit_test::label("boost::numeric::ublas::tensor_dynamic::operator=(...)")
+    *boost::unit_test::description("Testing dynamic tensor operator=(...)")
+)
+BOOST_FIXTURE_TEST_CASE_TEMPLATE(test_tensor_dynamic,
+    TestTupleType,
+    boost::numeric::ublas::test_types,
+    boost::numeric::ublas::tuple_fixture_tensor_dynamic<TestTupleType>
+){
+    namespace ublas = boost::numeric::ublas;
+    using value_type = typename TestTupleType::first_type;
+    using layout_type = typename TestTupleType::second_type;
+    using tensor_t = ublas::tensor_dynamic<value_type,layout_type>;
+    using strides_t = typename tensor_t::strides_type;
+    using fixture_type = boost::numeric::ublas::tuple_fixture_tensor_dynamic<TestTupleType>;
+    
+    // FIXME: Enable this test after the expressions are fixed
+    // BOOST_TEST_CONTEXT("[Dynamic Tensor Assignment Operator] assigning tensor expression"){
+        
+    //     auto const e = fixture_type::t32.extents();
+
+    //     auto const a = tensor_t(e, value_type(1));
+    //     BOOST_REQUIRE_EQUAL ( a.size(), 6ul);
+    //     BOOST_REQUIRE_EQUAL ( a.rank(), 2ul);
+    //     BOOST_REQUIRE       (!a.empty());
+
+    //     auto const three = value_type(3);
+    //     auto const expr = a + three * a;
+        
+    //     auto t = tensor_t(e);
+    //     t = expr;
+
+    //     BOOST_REQUIRE_EQUAL ( t.size(), 6ul);
+    //     BOOST_REQUIRE_EQUAL ( t.rank(), 2ul);
+    //     BOOST_REQUIRE       (!t.empty());
+    //     BOOST_CHECK_NO_THROW( (void)t.at(3ul));
+    //     BOOST_CHECK_THROW   ( (void)t.at(0,1,1),   std::invalid_argument);
+    //     BOOST_CHECK_THROW   ( (void)t.at(0,1,2),   std::invalid_argument);
+    //     BOOST_CHECK_THROW   ( (void)t.at(25ul),    std::out_of_range);
+    //     BOOST_CHECK_THROW   ( (void)t.at(5,2),     std::out_of_range);
+        
+    //     BOOST_REQUIRE ( t.extents() == e );
+    //     auto const strides = (std::is_same_v<layout_type,ublas::layout::first_order> ? strides_t{1, 3} : strides_t{2ul, 1ul});
+    //     BOOST_REQUIRE_EQUAL_COLLECTIONS(std::begin(t.strides()), std::end(t.strides()), std::begin(strides), std::end(strides));
+        
+    //     auto const four = value_type(4);
+    //     for(auto i = 0ul; i < t.size(); ++i){
+    //         BOOST_TEST_CHECKPOINT("[Tensor Expression  Assignment] testing tensor element(" << i << ") = " <<t[i]);
+    //         BOOST_REQUIRE_EQUAL(t[i], four);
+    //     }
+    // }
+
+    BOOST_TEST_CONTEXT("[Dynamic Tensor Copy Assignment Operator] copying tensor"){
+        auto o = fixture_type::t32;
```

Comment:
There are two reasons:
- More tests -> slower runtime and the GitHub actions fails due to overuse of RAM.
- Because it constructs a new tensor and then it swaps them. Therefore, it has less surface area for testing, and we are already intensely tested constructors which, in turn, makes the tests redundant or unnecessary.